gridstatus/tests/test_nyiso.py

```diff
@@ -108,6 +105,10 @@ def test_get_load_month_range(self):
         df = self.iso.get_load(start="2023-04-01", end="2023-05-16")
         assert df.shape[0] >= 0
 
+    def test_get_load_historical(self):
+        # TODO: why does this not work more than 8 days in the past
+        super().test_get_load_historical(lookback_days=8)
```

Comment:
Should there be more than 8 days of historical load?

Reply:
it might have something to do with how nyiso has daily files for the last 10 days and the it switches to monthly archives: http://mis.nyiso.com/public/P-58Blist.htm
<img width="493" alt="image" src="https://github.com/kmax12/gridstatus/assets/371792/0e09e363-3c25-44f9-ae2c-6577399a6c6d">

Reply:
Should the source code be able to handle this, or is 7 days of past load okay?

Reply:
it would be worth fixing. every nyiso endpoint pretty much goes through the same download function that is supposed to be handling this: https://github.com/kmax12/gridstatus/blob/b2894a69d7ae127ae2c441aceaa295bd70d332d6/gridstatus/nyiso.py#L689

Reply:
The function is downloading the archived file (http://mis.nyiso.com/public/csv/pal/20240101pal_csv.zip) but that archive doesn't contain data more than ~9 days in the past.

<img width="205" alt="Screenshot 2024-01-23 at 11 50 27" src="https://github.com/kmax12/gridstatus/assets/17484616/966fc601-0a4e-4bc5-ba40-2e2941f9e710">

Reply:
To clarify, this issue only seems to affect that one archive zip file (20240101). The other archives have all the days in the month.